coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,790 @@
+	if err := coord.ExecuteJob(metadata.GetJobKey(sourceID)); err != nil {
+		return err
+	}
+	if err := coord.ExecuteJob(metadata.GetJobKey(featureID)); err != nil {
+		return err
+	}
+
+	resourceTable, err := onlineStore.GetTable(featureName, "")
+	if err != nil {
+		return err
+	}
+	for _, record := range testOfflineTableValues {
+		value, err := resourceTable.Get(record.Entity)
+		if err != nil {
+			return err
+		}
+		if !reflect.DeepEqual(value, record.Value) {
+			return fmt.Errorf("Feature value did not materialize")
+		}
+	}
+	featureFirstTimestamp, err := client.GetFeatureVariant(context.Background(), metadata.NameVariant{Name: featureName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get training set")
+	}
+	oldTimestamp := featureFirstTimestamp.LastUpdated()
+	if err := UpdateOriginalPostgresTable(originalTableName); err != nil {
+		return err
+	}
+	go func() {
+		if err := coord.WatchForUpdateEvents(); err != nil {
+			logger.Errorf("Error watching for new update events: %v", err)
+		}
+	}()
+	time.Sleep(70 * time.Second)
+	jobClient, err := runner.NewKubernetesJobClient(runner.GetCronJobName(featureID), runner.Namespace)
+	if err != nil {
+		return err
+	}
+	cronJob, err := jobClient.GetCronJob()
+	if err != nil {
+		return err
+	}
+	lastExecutionTime := cronJob.Status.LastSuccessfulTime
+	if lastExecutionTime.IsZero() {
+		return fmt.Errorf("job did not execute in time")
+	}
+	resourceTable, err = onlineStore.GetTable(featureName, "")
+	if err != nil {
+		return err
+	}
+	for _, record := range testOfflineTableUpdateValues {
+		value, err := resourceTable.Get(record.Entity)
+		if err != nil {
+			return err
+		}
+		if !reflect.DeepEqual(value, record.Value) {
+			return fmt.Errorf("Feature table did not update")
+		}
+	}
+	featureSecondTimestamp, err := client.GetFeatureVariant(context.Background(), metadata.NameVariant{Name: featureName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get online data")
+	}
+	newTimestamp := featureSecondTimestamp.LastUpdated()
+	if !oldTimestamp.Before(newTimestamp) {
+		return fmt.Errorf("Resource update not signaled in metadata")
+	}
+	return nil
+}
+
+func testScheduleTransformation() error {
+	if err := runner.RegisterFactory(string(runner.CREATE_TRANSFORMATION), runner.CreateTransformationRunnerFactory); err != nil {
+		return fmt.Errorf("Failed to register training set runner factory: %v", err)
+	}
+	defer runner.UnregisterFactory(string(runner.CREATE_TRANSFORMATION))
+	logger := zap.NewExample().Sugar()
+	client, err := metadata.NewClient(metadata_addr, logger)
+	if err != nil {
+		return fmt.Errorf("Failed to connect: %v", err)
+	}
+	defer client.Close()
+	etcdConnect := fmt.Sprintf("%s:%s", etcdHost, etcdPort)
+	cli, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdConnect}, Username: "root",
+		Password: "secretpassword"})
+	if err != nil {
+		return err
```

Comment:
Can you just make a function that contains all this setup code? Its repeated a lot and takes up unneccessary space